### ISSUE TITLE
Simple GADT definition rejected

### ISSUE THREAD

**Issue by treeowl** [augustss/MicroHs#71]
The following code is accepted by GHC:

```haskell
module Narty where
import Data.Kind (Type)

type Nart :: (Type -> Type) -> Type -> Type -> Type
data Nart node a t where
  Z :: Nart node a a
  S :: Nart node a t -> Nart node a (node t)
```

MicroHs tells me `mhs: error: "./Narty.hs": line 7, col 38: kind error: cannot unify Type and (_a30 -> _a31)`.
I *believe* the code to be entirely sensible.

**Comment by treeowl** [comment]
This even happens for plain algebraic datatypes in GADT syntax:

```haskell
data Bar f a where
  Bar :: f a -> Bar f a
```

**Comment by augustss** [comment]
Type equality wasn't polykinded.  Fixed in 9450806259eb31116552c4b1b173ee68071d0b6d
